Source/WebCore/rendering/svg/LegacyRenderSVGShape.cpp

```diff
@@ -87,12 +87,11 @@ void LegacyRenderSVGShape::fillShape(GraphicsContext& context) const
 void LegacyRenderSVGShape::strokeShape(GraphicsContext& context) const
 {
     ASSERT(m_path);
-    Path* usePath = m_path.get();
 
     if (hasNonScalingStroke())
-        usePath = nonScalingStrokePath(usePath, nonScalingStrokeTransform());
-
-    context.strokePath(*usePath);
+        context.strokePath(nonScalingStrokePath(path()));
```

Comment:
Not sure if the nonScalingStrokePath() helper is worth it, since you need access to both the Path and the AffineTransform nonScalingTransform, in most places.

```
context.strokePath(nonScalingStrokeTransform().mapPath(path()));
```

seems quite readable as-is.